src/vdp_scanner.py

```diff
@@ -0,0 +1,296 @@
+    -t, --domain-csv=DOMAIN_CSV  Filename to use for Domain (TLD) results.
+    -p, --path-to-chrome=PATH    Path to the serverless-chrome binary being used
+                                 [default: /usr/local/bin/serverless-chrome]
+"""
+
+# Standard Python Libraries
+from collections import defaultdict
+import csv
+from datetime import datetime
+import logging
+from os.path import join as path_join
+from typing import Any, Dict, List, NamedTuple, Optional, Tuple
+from urllib.parse import urlparse, urlunparse
+
+# Third-Party Libraries
+import docopt
+import requests
+import urllib3
+
+# cisagov Libraries
+from hash_http_content import UrlHasher, UrlResult
+
+
+class DomainResult(NamedTuple):
+    """Structured format for a domain check result."""
+
+    domain: str
+    agency: str
+    organization: str
+    security_contact: str
+    visited_url: str
+    is_redirect: bool
+    vdp_present: bool
+
+
+class VdpScanner:
+    """Class to handle scanning and outputting the results of any scans."""
+
+    agency_csv_header = [
+        "Agency",
+        "Total Domains",
+        "Domains with Security Contact Listed",
+        "Domains with Organization Listed",
+        "Domains with Matching Organization and Agency",
+        "Domains with Published VDP",
+    ]
+
+    domain_csv_header = [
+        "Domain",
+        "Agency",
+        "Organization",
+        "Security Contact Email",
+        "Visited URL",
+        "Was it Redirected",
+        "VDP is Published",
+    ]
+
+    def __init__(self, hasher: UrlHasher):
+        """Initialize variables and perform setup."""
+        self._hasher = hasher
+        file_date = datetime.utcnow().strftime("%Y-%m-%d")
+        self.agency_csv = f"agency_results_{file_date}.csv"
+        self.domain_csv = f"domain_results_{file_date}.csv"
+        self.output_directory = "host_mount"
+
+        self.agency_results: defaultdict = defaultdict(
+            lambda: {k: 0 for k in self.agency_csv_header[1:]}
+        )
+
+        self.domain_results: List[Dict[str, Any]] = []
+
+    @staticmethod
+    def _log_vdp_failure(domain: str, err: Exception) -> None:
+        """Log failure information during check_for_vdp() execution."""
+        logging.warning("Unable to retrieve hash for '%s'", domain)
+        logging.debug("Caught %s", type(err).__name__)
+        logging.debug(err)
+
+    def check_for_vdp(self, domain: str) -> Tuple[str, bool, bool]:
+        """Check for a VDP at the given domain and return the relavent information."""
+        url = urlparse(f"https://{domain}/vulnerability-disclosure-policy")
+        result: Optional[UrlResult] = None
+
+        # Try with HTTPS first
+        try:
+            result = self._hasher.hash_url(urlunparse(url))
+        # If there is a TLS issue, try running it without verifying
+        except requests.exceptions.SSLError:
+            logging.warning("Trying '%s' without TLS verification", domain)
```

Comment:
Use a consistent format for the fallback warnings
```suggestion
            logging.warning("Falling back to HTTPS without TLS verification for '%s'", domain)
```